custom_components/readme/__init__.py

```diff
@@ -185,50 +185,84 @@ async def service_generate(call):
 
     hass.services.async_register(DOMAIN, "generate", service_generate)
 
-def get_hacs_components(hass):
-    base = hass.config.path()
-    keys = []
-    hacs_components = []
-
-    for file in os.listdir(f"{base}/.storage/hacs"):
-        if file.endswith(".hacs"):
-            keys.append(os.path.splitext(file)[0])
-
-    if os.path.exists(f"{base}/.storage/hacs.repositories"):
-        with open(f"{base}/.storage/hacs.repositories", "r") as repositories:
-            content = repositories.read()
-            content = json.loads(content)
-        
-        for key in keys:
-            repository = content["data"][key]
 
-            hacs_components.append(
+def get_hacs_components(hass):
+    try:
+        from custom_components.hacs.share import get_hacs
+    except ImportError:
+        return []
+
+    hacs = get_hacs()
+    repositories = hacs.repositories
+    installed = []
+
+    for repo in repositories:
+        if repo.data.installed:
+            installed.append(
```

Comment:
use `installed.append(repo.to_json())` instead, that is more dynamic

Reply:
Looks like this is actually `repo.data.to_json()`  Just pushed an update with this.

Reply:
Shows up on my history, but looks like github may be having some issues on the backend.

https://github.com/steve-gombos/readme/commit/f515461fd691047c335e7d6bb1b7bdc36458b35e

Reply:
Right, my bad. xxx.data.to_json() is the correct way
GitHub is having issues ATM, if it does not update the PR in an hour or so, just push an empty commit ty retrigger the sync

Reply:
Looks like it showed up.